EventFilter/DTRawToDigi/plugins/DTuROSRawToDigi.cc

```diff
@@ -0,0 +1,486 @@
+
+
+	  int wheelId, stationId, sectorId, slId,layerId, cellId;
+	  if ( ! mapping->readOutToGeometry(dduId, rosId, robId, tdcId, tdcChannel,
+					    wheelId, stationId, sectorId, slId, layerId, cellId)) {
+
+	    DTWireId detId = DTWireId(wheelId, stationId, sectorId, slId, layerId, cellId);
+	    int wire = detId.wire();
+
+	    DTDigi digi(wire, tdcTime, hitOrder[channelIndex.getCode()]);
+	    digis.insertDigi(detId.layerId(),digi);
+
+	  }
+
+	}
+        else if ( selector == 3 ) { // error word
+
+	  if (  debug_ ) edm::LogWarning("dturos_unpacker") << "Error word [" << std::dec << k << "] : " << std::hex << dataWord 
+							    << std::dec << " in slot " << slot << " in crate " << crate;
+
+	  int error  = ( dataWord >> posErrorSel3 ) & 0x1FFFFFFF; 
+	  rwords.seterror(error);
+
+	}
+
+
+	if ((dataWord & 0x1FFFFFFF) == 0x1FFFFFFF) continue;
+
+	if ( selector2 == 0 ) { // TDC word
+
+	  int tdcTime    = ( dataWord ) & 0x3FFF;       // positions   0 -> 13
+	  int tdcChannel = ( dataWord >> posTDCChannelSel2Null ) & 0x1F;   
+	  int tdcId      = ( dataWord >> posTDCIdSel2Null ) & 0x3;    
+	  int link       = ( dataWord >> posLinkSel2Null ) & 0x7F;   // positions  21 -> 27
+
+
+	  if (tdcTime == 16383) continue;
+
+	  int dummy = 0;
+
+	  bool tenDDU = ! mapping->readOutToGeometry(779, 7, 1, 1, 1,
+						     dummy, dummy, dummy, 
+						     dummy, dummy, dummy);
+
+	  int dduId = theDDU(crate, slotMap, link, tenDDU);
+	  int rosId = theROS(crate, slotMap, link);
+	  int robId = theROB(crate, slotMap, link);
+
+
+	  DTROChainCoding channelIndex(dduId, rosId, robId, tdcId, tdcChannel);
+	  if (hitOrder.find(channelIndex.getCode()) == hitOrder.end()) hitOrder[channelIndex.getCode()] = 0;
+	  else hitOrder[channelIndex.getCode()]++;
+
+
+	  int wheelId, stationId, sectorId, slId,layerId, cellId;
+	  if ( ! mapping->readOutToGeometry(dduId, rosId, robId, tdcId, tdcChannel,
+					    wheelId, stationId, sectorId, slId, layerId, cellId)) {
+
+	    DTWireId detId = DTWireId(wheelId, stationId, sectorId, slId, layerId, cellId);
+	    int wire = detId.wire();
+
+	    DTDigi digi(wire, tdcTime, hitOrder[channelIndex.getCode()]);
+	    digis.insertDigi(detId.layerId(),digi);
+
+	  }
+
+	}
+        else if ( selector2 == 1 ) { // error word
+
+	  if (  debug_ ) edm::LogWarning("dturos_unpacker") << "Error word [" << std::dec << k << "] : " << std::hex << dataWord 
+							    << std::dec << " in slot " << slot << " in crate " << crate;
+
+	  int error  = ( dataWord ) & 0x1FFFFFFF;       // positions   0 -> 28
+	  rwords.seterror(error);
+
+	}
+
+      }
+
+    }
+
+
+    ++DTuROSiterator;
+    dataWord  = (*DTuROSiterator); // Trailer AMC
+
+
+    rwords.settrailer(dataWord);
+    fwords.setuROS(slot, rwords);
+
+  }  // end for-loop container content
+
+
+  return;
+}
+
+
+int DTuROSRawToDigi::theDDU(int crate, int slot, int link, bool tenDDU) {
+
+  int ros = theROS(crate,slot,link); 
+  
+  int ddu = 772;
+
+  if (crate == 1368) { ddu = 775; }
```

Comment:
Is ```crate=1368``` still allowed?

Reply:
Yes, it is a "just in case". The hardware conditions are still pending somehow, sorry. I have included all your other comments.
Thanks for your patience...